bands/site_sections/bands.py

```diff
@@ -9,57 +9,49 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, band_id, message='', **params):
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
```

Comment:
There are a TON of these manual if/else statements throughout this file.  The reason for this was that at the time I was unable to use the ``@validation`` API and just run ``check(band)``.  However, now that we've split everything out into its own model, we can start marking all of this stuff as ``required`` and/or write validation functions in ``model_checks.py`` or wherever.  That would drastically simplify this file and we should definitely do it if we have time.

Since what I just suggested is basically "someone needs to clean up Eli's crappy old code that he wrote in a hurry and didn't have time to get right", feel free to ignore this suggestion and let me handle it the next time I'm making changes to this plugin myself :)